src/clib/pio_darray.c

```diff
@@ -647,10 +648,10 @@ int PIOc_write_darray(int ncid, int varid, int ioid, PIO_Offset arraylen, void *
 
     /* If the type of the var doesn't match the type of the
      * decomposition, return an error. */
-    if (iodesc->piotype != vdesc->pio_type)
-        return pio_err(ios, file, PIO_EINVAL, __FILE__, __LINE__);
-    pioassert(iodesc->mpitype_size == vdesc->mpi_type_size, "wrong mpi info",
-              __FILE__, __LINE__);
```

Comment:
Should we check that the size of the type is the same? Because if you do this with a 1-byte and a 4-byte type, won't hilarious confusion ensue?

Reply:
Yes - problems will occur if you try to use an iodesc defined for one type size with a variable of another type size.   I am working on a test.

Reply:
It's not possible as far as I can tell to capture this error - in the call 
 PIOc_write_darray(ncid, varid, ioid, arraylen, test_data, fillvalue)

the type of test_data must be of the same type_size as the type the ioid was declared with, but there is no way to test for that since the type in varid is the files type and not necessarily the type of the variable in memory.